File: testProjects/iOS/Appium-UIAutomation/Appium-UIAutomation/Utility.swift
//
//  Utility.swift
//  Appium-UIAutomation
//
//  Created by Anjum Shrimali on 4/5/17.
//  Copyright © 2017 IshiSystems. All rights reserved.
//

import Foundation

class Utility {
    var displayName:String?, username:String?, password:String?
    
    private init() { }
    
    static let shared = Utility()
    
    func register(displayName:String, username:String, password:String) -> Void {
        self.displayName = displayName
        self.username = username
        self.password = password
    }
    
    func login(username:String, password:String) -> Bool {
        if let usr = self.username, let pwd = self.password, usr == username && pwd == password {
            return true
        }
        
        return false
    }
    
    
}

File: testProjects/Android/Appium-UIAutomation/app/src/main/java/com/ishisystems/appium_uiautomation/commons/Utility.java
package com.ishisystems.appium_uiautomation.commons;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;

/**
 * Created by anjum.shrimali on 4/5/17.
 */

public class Utility {
    public static void showInfoDialog(String msg, final Runnable callBack, Context mContext) {
        AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
        builder.setMessage("Info");
        builder.setMessage(msg);
        builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {

            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
                if (callBack != null)
                    callBack.run();
            }
        });
        builder.show();
    }

    private static String displayName = "N/A", username = "", password = "";

    public static void register(String displayName, String username, String password) {
        Utility.displayName = displayName;
        Utility.username = username;
        Utility.password = password;
    }

    public static boolean login(String username, String password) {
        if (username.equals(Utility.username) && password.equals(Utility.password))
            return true;
        else
            return false;
    }

    public static String getDisplayName() {
        return Utility.displayName;
    }
}
